syncer/service_test.go

```diff
@@ -0,0 +1,155 @@
+package syncer
+
+import (
+	"context"
+	"io"
+	"log"
+	"net"
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/syncer/proto"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+const bufSize = 1024 * 1024
+
+func newMockGrpcClient(t *testing.T, service *syncPeerService) proto.SyncPeerClient {
+	t.Helper()
+
+	lis := bufconn.Listen(bufSize)
+	s := grpc.NewServer()
+	proto.RegisterSyncPeerServer(s, service)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	ctx := context.Background()
+	conn, err := grpc.DialContext(ctx, "bufnet",
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(
+			func(ctx context.Context, address string) (net.Conn, error) {
+				return lis.Dial()
+			},
+		),
+	)
```

Comment:
Please note that you are not "mocking" anything here, you're literally instantiating a gRPC client.

Why could you just write a mock client implementation, and tested it the same way we test `discovery` and `identity` mocks?

Reply:
I don't see what you're trying to say. Do you know `bufnet`? Or do you want to create mock of gRPC?

Reply:
I don't know what's the benefit of mocking here

Reply:
If we define it as mock, what do we test? We test mock returns predefined value? Is this test?

Reply:
The objective of the tests in this file is to check (1) server can get info by correct way (2) gRPC encoding/decoding is done correctly

Reply:
Yes, depends on what you're mocking:

* If you're mocking a server, then your client code is "real"
* If you're mocking a client, then your server code is "real"

This approach you took doesn't take up time (it's only a few seconds), but I just wanted to note we could've done this without `bufnet` and real gRPC connections

Reply:
Then what is our conclusion. Do you just want me to rename?

Reply:
ping @zivkovicmilos

Reply:
@Kourin1996 

No need, just wanted to have this discussion 🙂